client/src/components/ui/dropdown.tsx

```diff
@@ -0,0 +1,180 @@
+          tabIndex={-1}
+        >
+          {options.map((option, index) => (
+            <li
+              key={index}
+              value={option}
+              className={`${valueStyle} select-none rounded-penni-border px-2 py-2 hover:cursor-pointer hover:bg-penni-grey-inactive`}
+              role="menuitem"
+              tabIndex={-1}
+              onClick={() =>
+                onChange({
+                  target: { value: option },
+                } as React.ChangeEvent<HTMLEventTargetElement>)
+              }
+            >
+              {option}
+            </li>
+          ))}
+        </ul>
+      </div>
+    </div>
+  );
+}
+
+/**
+   * Dropdown component in bidders side market page with a list of options.
+   
+   * @param props - The properties for the DropdownInput component.
+   * @param props.value - Value of the selected option, tracked by `react.setState()`.
+   * @param props.options - List of possible option for that category, displayed in dropdown.
+   * @param props.onChange - Called when the selected value changes, should update `value`.
+   * @param props.label - Label for the field, display "All ...".
+   *
+   * @returns dropdown component to select different categories.
+   *
+   * @example
+   * // Example usage:
+   * // initial state is '', display all tasks, them use onChange function to get selected category that can be used to send query to database
+   * const [selectedValue, setSelectedValue] = useState('');
+   * <MarketDropdown
+   *   value={selectedValue}
+   *   options={['C', 'Python', 'HTML']}
+   *   onChange={(e) => setSelectedValue(e.target.value)}
+   *   label="All Language"
+   * />
+   */
+export function MarketDropdown({
+  value,
+  options,
+  onChange,
+  label,
+}: MarketDropdownProps) {
+  const [menuId] = useState(uniqueId());
+  const [isExpanded, setExpanded] = useState(false); // isSelect in text inputs
+
+  function handleOnChange(e: React.ChangeEvent<HTMLEventTargetElement>) {
+    setExpanded(false);
+    onChange(e);
+  }
+  function handleOnClick() {
+    setExpanded(!isExpanded);
+  }
+  const containerStyle =
+    `duration-50 flex flex-row h-9 w-full items-center rounded-penni-border px-3  border transition-all ease-out` +
+    ` ${isExpanded ? selectedStyle : deselectedStyle} `;
+  const optionList = [label, ...options];
+  return (
+    <div className="relative w-36">
+      <DropdownButton
+        id={menuId}
+        style={containerStyle}
+        onClick={handleOnClick}
+        caretWidth={10}
+      >
+        {value === "" ? (
+          <label htmlFor={menuId} className={` ${labelStyle} text-left`}>
+            {label}
+          </label>
+        ) : (
+          <span className={`${valueStyle} text-left`}>{value}</span>
+        )}
+      </DropdownButton>
+
+      {isExpanded && (
+        <DropdownMenu
+          menuId={menuId}
+          options={optionList}
+          onChange={handleOnChange}
+        />
+      )}
```

Comment:
Why is `isExpanded` logic here? Should be in `DropdownButton`, all that's needed in `MarketDropdown` is `<DropdownButton>{elements here...}</DropdownButton>`

Reply:
Ok I know what you mean now I'll move these to <DropdownButton>